sqlalchemy_dremio/base.py

```diff
@@ -0,0 +1,266 @@
+                self.process(join.right, asfrom=True) + " ON " + \
+                self.process(join.onclause) + ')')
+
+    def visit_extract(self, extract, **kw):
+        field = self.extract_map.get(extract.field, extract.field)
+        return 'DATEPART("%s", %s)' % \
+                    (field, self.process(extract.expr, **kw))
+
+class DremioDDLCompiler(compiler.DDLCompiler):
+    def get_column_specification(self, column, **kwargs):
+        if column.table is None:
+            raise exc.CompileError(
+                            "dremio requires Table-bound columns "
+                            "in order to generate DDL")
+
+        colspec = self.preparer.format_column(column)
+        seq_col = column.table._autoincrement_column
+        if seq_col is column:
+            colspec += " AUTOINCREMENT"
+        else:
+            colspec += " " + self.dialect.type_compiler.process(column.type)
+
+            if column.nullable is not None and not column.primary_key:
+                if not column.nullable or column.primary_key:
+                    colspec += " NOT NULL"
+                else:
+                    colspec += " NULL"
+
+            default = self.get_column_default_string(column)
+            if default is not None:
+                colspec += " DEFAULT " + default
+
+        return colspec
+
+    def visit_drop_index(self, drop):
+        index = drop.element
+        self.append("\nDROP INDEX [%s].[%s]" % \
+                        (index.table.name,
+                        self._index_identifier(index.name)))
+
+class DremioIdentifierPreparer(compiler.IdentifierPreparer):
+    reserved_words = compiler.RESERVED_WORDS.copy()
+    reserved_words.update(['value', 'text'])
+    def __init__(self, dialect):
+        super(DremioIdentifierPreparer, self).\
+                __init__(dialect, initial_quote='[', final_quote=']')
+
+
+
+class DremioDialect(default.DefaultDialect):
+    colspecs = {
+        types.Unicode: AcUnicode,
+        types.Integer: AcInteger,
+        types.SmallInteger: AcSmallInteger,
+        types.Numeric: AcNumeric,
+        types.Float: AcFloat,
+        types.DateTime: AcDateTime,
+        types.Date: AcDate,
+        types.String: AcString,
+        types.LargeBinary: AcBinary,
+        types.Boolean: AcBoolean,
+        types.Text: AcText,
+        types.CHAR: AcChar,
+        types.TIMESTAMP: AcTimeStamp,
+    }
+    name = 'dremio'
+    supports_sane_rowcount = False
+    supports_sane_multi_rowcount = False
+
+    poolclass = pool.SingletonThreadPool
+    statement_compiler = DremioCompiler
+    ddl_compiler = DremioDDLCompiler
+    preparer = DremioIdentifierPreparer
+    execution_ctx_cls = DremioExecutionContext
+
+    @classmethod
+    def dbapi(cls):
+        import pyodbc as module
+        return module
+
+    def create_connect_args(self, url):
+        opts = url.translate_connect_args()
+        connectors = ["Driver={Dremio Driver (*.mdb)}"]
```

Comment:
This seems to be Access specific cruft -- there be no mdb's in Dremio.
The`Driver` should be set to the actual ODBC driver name for the given OS/platform.

```python
driver_for_platf = {
 'Linux 64bit': 'Dremio ODBC Driver 64-bit',
 'Linux 32bit': 'Dremio ODBC Driver 32-bit',
 'Windows' : 'Dremio Connector',
 'Darwin': 'Dremio ODBC Driver'
}
platf = platform.system() + ' ' + (platform.architecture()[0] if platform.system() == 'Linux' else '')
drv = driver_for_platf[platf]
connectors = ["Driver=%s" % drv]
```